recirq/quantum_chess/quantum_board.py

```diff
@@ -808,8 +809,6 @@ def do_move(self, m: move.Move) -> int:
             tqubit2 = bit_to_qubit(tbit2)
             self.add_entangled(squbit, tqubit, tqubit2)
             self.circuit.append(qm.split_move(squbit, tqubit, tqubit2))
-            self.state = set_nth_bit(sbit, self.state, False)
-            self.unhook(squbit)
```

Comment:
Maybe we could change this to only unhook if the target squares were unentangled before moving?

Otherwise, this will leave a lot of entangled squares on the board which will be a source of unmitigated noise.

Reply:
OK, I added the optimization to unhook in the unentangled 100 state.

In general there seems to be a lot of low-hanging fruit for this sort of qubit-saving operations (e.g. a path qubit is not needed for a split slide if there are no obstructing qubits on one path).

Reply:
We should consider making a bug for these sorts of optimizations.  Seems like something a 20%er could easily tackle.